src/firmware/app/control/trajectory_planner.h

```diff
@@ -181,123 +213,177 @@ static void app_trajectory_planner_generateConstArclengthTrajectoryPositions(
- * @param arc_segment_length [in] The length of each arc length segment in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point in
- * m/s^2
- * @param max_allowable_speed [in] The max allowable speed at any point in m/s
+ * @param trajectory [in/out] The trajectory to be modified
  *
- * */
-void app_trajectory_planner_generateForwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration,
-    const float max_allowable_speed);
+ * @param segment_lengths [in] The length of each segment in the trajectory
+ *
+ * @param desired_segment_duration [in] The desited segment duration in seconds
+ *
+ * @param trajectory_index [in] The index of end-point of the segment to be modified
+ *
+ * @param rebalance_angular [in] Boolean to select if the angular or linear segment is to
+ * be modified. rebalance_angular == true means that the angular segment will be modified
+ * @return
+ */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_rebalanceAngularAndLinearTrajectorySegmentsForEquivalentDuration(
+    PositionTrajectory_t* trajectory, TrajectorySegment_t* segment_lengths,
+    const float desired_segment_duration, unsigned int trajectory_index,
+    bool rebalance_angular);
+
+/**
+ * This function generates a velocity trajectory that corresponds to the time-optimal
+ * velocity to follow a specified path. This profile is based on the input position
+ * trajectory
+ *
+ * @param position_trajectory [in] The position trajectory to build the velocity
+ * trajectory from
+ * @param velocity_trajectory [out] The velocity trajectory that corresponds to the
+ * time-optimal velocity to follow a specified path
+ */
+void app_trajectory_planner_generateVelocityTrajectory(
+    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory);
 
 /**
- * Modifies a forwards continuous velocity profile to also be backwards continuous based
- * on the input parameters.
+ * Returns a planned trajectory with the list of guarantees based on the assumptions below
  *
- * NOTE: This function does not break forwards continuity of the trajectory
+ *  Key assumptions & guarantees of this planner are:
+ *  - Trajectories are time-optimal assuming INFINITE JERK capability of the robot
+ *
+ *  - No speed on the trajectory is larger than the 'max_allowable_linear_speed' parameter
+ *
+ *  - No acceleration value between points on the trajectory can be larger than the
+ * 'max_allowable_linear_acceleration' input parameter
+ *
+ *  - Assuming the grip-limit of the robot IS THE SAME AS THE MAX ACCELERATION then at no
+ * point on the path can the sum of centripetal and acceleration force be greater than
+ * 'max_allowable_linear_acceleration'
```

Comment:
bit of a nit; but for bullet points like these could we make sure subsequent lines line up with the first? ie.
``` suggestion
 *  - Assuming the grip-limit of the robot IS THE SAME AS THE MAX ACCELERATION then at no
 *    point on the path can the sum of centripetal and acceleration force be greater than
 *    'max_allowable_linear_acceleration'
```